packages/sdk-react/src/components/userRoles/UserRoleDetails/ExistingUserRoleDetails/ExistingUserRoleDetails.tsx

```diff
@@ -0,0 +1,315 @@
+import React from 'react';
+
+import { useDialog } from '@/components';
+import { getPermissionToLabelMap } from '@/components/userRoles/consts';
+import { MoniteStyleProvider } from '@/core/context/MoniteProvider';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import {
+  RoleResponse,
+  RootSchema,
+  PayableActionEnum,
+  ActionEnum,
+  PermissionEnum,
+  CommonSchema,
+  PayableSchema,
+} from '@monite/sdk-api';
+import {
+  CheckRounded as CheckRoundedIcon,
+  CloseRounded as CloseRoundedIcon,
+  Close as CloseIcon,
+  OpenInNew as OpenInNewIcon,
+} from '@mui/icons-material';
+import {
+  styled,
+  Box,
+  DialogTitle,
+  DialogContent,
+  Divider,
+  IconButton,
+  Typography,
+  Link,
+  Table,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+  tableCellClasses,
+} from '@mui/material';
+import { GridRenderCellParams, GridValueGetterParams } from '@mui/x-data-grid';
+
+type Actions = {
+  [key in ActionEnum | PayableActionEnum]?: boolean;
+};
+
+type PermissionRow = Actions & {
+  name?: (CommonSchema | PayableSchema)['object_type'];
+};
+
+interface ExistingUserRoleDetailsProps {
+  /** Role to be displayed */
+  role: RoleResponse;
+}
+
+const validatePermission = (permission?: PermissionEnum) => {
+  return (
+    permission === PermissionEnum.ALLOWED ||
+    permission === PermissionEnum.ALLOWED_FOR_OWN
+  );
+};
+
+const normalizePermissions = (objects: RootSchema[]) => {
+  return objects.map((object) => {
+    const permission: PermissionRow = {};
+
+    if (object.object_type) permission.name = object.object_type;
+
+    object.actions?.forEach((action) => {
+      if (action.action_name) {
+        permission[action.action_name] = validatePermission(action.permission);
+      }
+    });
+
+    return permission;
+  });
+};
+
+const renderPermissionCell = (params: GridRenderCellParams<PermissionRow>) => {
+  const actionValue = params.row[params.field as keyof Actions];
+
+  if (actionValue === undefined) {
+    return null;
+  } else if (actionValue) {
```

Comment:
It is not necessary to use `else` in this case.

Reply:
removed